ci.sh

```diff
@@ -76,12 +76,21 @@ rebuild() {
 
 # test druntime, phobos, dmd
 test() {
+    run_unit_tests
     test_dub_package
     make -j$N -C ../druntime -f posix.mak MODEL=$MODEL unittest
     make -j$N -C ../phobos -f posix.mak MODEL=$MODEL unittest
     test_dmd
 }
 
+run_unit_tests() {
+    source ~/dlang/*/activate # activate host compiler
```

Comment:
Any reason why you didn't use the `dub_package` folder?

`test_dub_package` does two important things (that you don't do here)
- doesn't run for GDC (as there's no `dub` installed for GDC)
- runs the dub package tests for the host compiler _and_ currently built compiler

Reply:
> Any reason why you didn't use the dub_package folder?

Yes, the `dub_package` was originally intended to test the Dub package, not the front end, not the compiler. That is, basically make sure that all necessary files are included and the correct flags are used. I didn't have to use Dub for the unit tests, I could just have used a simple shell script that collects all the files, compiles them and run the tests. But I consider Dub to be more portable and it will do that all that automatically.

> doesn't run for GDC (as there's no dub installed for GDC - that's why SemaphoreCI is failing)

That's unfortunate. Can we add it?

> runs the dub package tests for the host compiler and currently built compiler

Do you think that is important? Are we doing that for the other tests (not the Dub package)?

This need to be moved from the `ci.sh` file to the makefile and `run.d`. These tests should always be run, also when running the tests locally. This is intended as an alternative to the existing `compilable`, `fail_compilation` and `runnable`. If we can't use Dub for that, I'll have to come up with something else.

Reply:
I would need some help to add it to `run.d` and the makefile.